airflow/contrib/operators/dataproc_operator.py

```diff
@@ -928,3 +930,125 @@ def execute(self, context):
         job.set_job_name(self.job_name)
 
         hook.submit(hook.project_id, job.build(), self.region)
+
+
+class DataprocWorkflowTemplateInstantiateOperator(BaseOperator):
+    """
+    Instantiate a WorkflowTemplate on Google Cloud Dataproc. The operator will wait
+    until the WorkflowTemplate is finished executing.
+
+    Please refer to:
+    https://cloud.google.com/dataproc/docs/reference/rest/v1beta2/projects.regions.workflowTemplates/instantiate
+
+    :param template_id: The id of the template.
+    :type template_id: string
+    :param project_id: The ID of the google cloud project in which
+        the template runs
+    :type project_id: string
+    :param region: leave as 'global', might become relevant in the future
+    :type region: string
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+    :type gcp_conn_id: string
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have domain-wide
+        delegation enabled.
+        :type delegate_to: string
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 template_id,
+                 project_id,
+                 region='global',
+                 gcp_conn_id='google_cloud_default',
+                 delegate_to=None,
+                 *args,
+                 **kwargs):
+
+        super(DataprocWorkflowTemplateInstantiateOperator, self).__init__(*args, **kwargs)
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.template_id = template_id
+        self.project_id = project_id
+        self.region = region
+
+    def execute(self, context):
+        self.log.info('Instantiating Template: %s', self.template_id)
+        hook = DataProcBetaHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to
+        )
+        service = hook.get_conn()
+
+        name = ('projects/%s/regions/%s/workflowTemplates/%s' %
+                (self.project_id, self.region, self.template_id))
+        request_id = str(uuid.uuid1())
+
+        operation = service.projects().regions().workflowTemplates().instantiate(
+            name=name, body={'instanceId': request_id}).execute()
+
+        self.log.info(
+            'WorkflowTemplate instantiate operation name: %s', operation['name'])
+        hook.await(operation)
+
+
+class DataprocWorkflowTemplateInstantiateInlineOperator(BaseOperator):
```

Comment:
There's quite substantial code duplication between templateinstantiate and templateinstantiateinline operators, do you want to refactor and clean up a bit? I'll leave this to you.

Reply:
Could you suggest how this should be merged?